src/OpenTK.GLControl/GLControl.cs

```diff
@@ -119,6 +120,15 @@ public GLControl(GraphicsMode mode, int major, int minor, GraphicsContextFlags f
             InitializeComponent();
         }
 
+        /// <summary>
+        /// Gets a value indicating whether [failed to create OpenGL context].
+        /// So that the application stays running and able to recover.
+        /// </summary>
+        /// <value>
+        ///   <c>true</c> if [failed create context]; otherwise, <c>false</c>.
+        /// </value>
+        public bool FailedCreateContext { get; private set; }
```

Comment:
Checking "error codes" is a very C-like API. We'd prefer to stay with a more typical OO approach if possible - ideally we want to throw an exception, (but not in the constructor).

@Frassle  - reckon we could use an explicit initialize method or factory method here without breaking GLControl?

Reply:
Probably not, a flag actually makes the most sense here because the object is now valid even with an invalid context. But I would flip the logic of this flag to be positive, calling it something like "HasValidContext" or similar (generally easier to reason about truthy flags)

Reply:
Yeah, this is why I wanted to decouple it into an initialization method rather than throwing it in the constructor.

That way you still get the object, but you know the initialization didn't work.

Reply:
Yeah, this is why I wanted to decouple it into an initialization method rather than throwing it in the constructor.

That way you still get the object, but you know the initialization didn't work.

Reply:
Would need to check if this will work, since at the moment the actual init is in the OnHandleCreated overload. This is kind of OK, since you do want to create a new GLContext every time you recreate the WindowHandle.

Possible solution:  have GLControl init once explicitly by the caller and that Init can be rerun each time you enter OnHandleCreated. Of course you can only run Init when you have a WindowsHandle at the first time.